update async-synthesis-long-form-audio.md -yy (#45219)
Azure/azure-content: articles/cognitive-services/Speech-Service/quickstarts/text-to-speech/async-synthesis-long-form-audio.md

# Quickstart: Asynchronous synthesis for long-form audio in Python (Preview)

> In this quickstart, you'll use the Long Audio API to asynchronously convert text to speech, and retrieve the audio output from a URI provided by the service. This REST API is ideal for content providers that need to synthysize audio more than 10 minutes. For more information, see [Long Audio API](../../long-audio-api.md).

Reviewer: Hi @Yueying-Liu , how does someone know that the text file is longer than ten minutes? This isn't clear to me, whereas, the characters are something that they can measure.

Author: This REST API is ideal for content providers that need to synthesize audio more than 5,000 characters

Reviewer: Let's update the text to say that, since it's hard for a first time user to determine how much text makes up 10 minutes.

Reviewer: As you've stated below. Let's use the 5,000 characters, and we can state that it's approximately 10 minutes of audio.

> In this quickstart, you'll use the Long Audio API to asynchronously convert text to speech, and retrieve the audio output from a URI provided by the service. This REST API is ideal for content providers that need to synthysize audio more than 10 minutes. For more information, see [Long Audio API](../../long-audio-api.md).

Reviewer: ```suggestion
In this quickstart, you'll use the Long Audio API to asynchronously convert text to speech, and retrieve the audio output from a URI provided by the service. This REST API is ideal for content providers that need to synthesize audio from text greater than 5,000 character (or more than 10 minutes in length). For more information, see [Long Audio API](../../long-audio-api.md).
```

Author: It looks good.

### Test your code

Let's test what you've done so far. You'll need to update a few things in the request below:

* Replace `<your_key>` with your Speech service subscription key. This information is available in the **Overview** tab for your resource in the [Azure portal](https://aka.ms/azureportal).
* Replace `<region>` with the region where your Speech resource was created (for example: `eastus` or `westus`). This information is available in the **Overview** tab for your resource in the [Azure portal](https://aka.ms/azureportal).

Run this command:

```console
python voice_synthesis_client.py --voices -key <your_key> -region <Region>
```

> You will get the output like this:

Reviewer: ```suggestion
You'll see an output that looks like this:
```

Name: Microsoft Server Speech Text to Speech Voice (zh-CN, xxx), Description: xxx , Id: xxx, Locale: zh-CN, Gender: Female, PublicVoice: xxx, Created: 2019-08-26T04:55:39Z
```

## Prepare input files

Prepare an input text file. It can be either plain text or SSML text. For the input file requirements, see how to [prepare content for synthesis](https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/long-audio-api#prepare-content-for-synthesis).

## Convert text to speech

> After getting the input text file ready, add the code for speech synthesis to `voice_synthesis_client.py`:

Reviewer: ```suggestion
After preparing the input text file, add this code for speech synthesis to `voice_synthesis_client.py`:
```

### Test your code

> Let's try to make a request to synthesize text using your input file as a source. You'll need to update a few things in the request below:

Reviewer: ```suggestion
Let's make a request to synthesize text using your input file as the source. You'll need to update a few things in the request below:
```

```

> [!NOTE]
> If you have more than 1 input files, you will need to submit multiple requests. There are some limitations that needs to be aware. 
> * Client is allowed to submit up to **5** requests to server per second for each Azure subscription account. If it exceeds the limitation, client will get a 429 error code(too many requests). Please reduce the request amount per second
> * Server is allowed to run and queue up to **120** requests for each Azure subscription account. If it exceeds the limitation, server will return a 429 error code(too many requests). Please wait and avoid submitting new request until some requests are completed

> You will get the output like this:

Reviewer: ```suggestion
 see an output that looks like this:
```

Succeeded... Result file downloaded : xxxx.zip
```

> The result contains the input text and the audio output files generated by the service. These are downloaded as a zip.

Reviewer: ```suggestion
The result contains the input text and the audio output files that are generated by the service. You can download these files in a zip.
```

## Remove previous requests

> Server will keep up to **20,000** requests for each Azure subscription account. If your request amount exceeds the limitation, please remove previous requests before making new ones. If you don't remove existing request timely, you'll receive an error notification.

Reviewer: ```suggestion
The server will keep up to **20,000** requests for each Azure subscription account. If your request amount exceeds this limitation, please remove previous requests before making new ones. If you don't remove existing requests, you'll receive an error notification.
```

### Test your code

> Let's check your submitted requests before. You'll need to update a few things in the request below:

Reviewer: This is a bit unclear. Are you trying to say that the following request is used to check what requests had been made previously?

Author: yes. This action is to check all the synthesis request under user's Azure subscription.

Reviewer: How about this? (see below)

> Let's check your submitted requests before. You'll need to update a few things in the request below:

Reviewer: ```suggestion
Now, let's check to see what requests you've previously submitted. Before you continue, you'll need to update a few things in this request:
```

Author: it looks good.

* Replace `<your_key>` with your Speech service subscription key. This information is available in the **Overview** tab for your resource in the [Azure portal](https://aka.ms/azureportal).
* Replace `<region>` with the region where your Speech resource was created (for example: `eastus` or `westus`). This information is available in the **Overview** tab for your resource in the [Azure portal](https://aka.ms/azureportal).

Run this command:

```console
python voice_synthesis_client.py – syntheses -key <your_key> -region <Region>
```

> This will return a list of synthesis you've requested. You will get the output like this:

Reviewer: ```suggestion
This will return a list of synthesis requests that you've made. You'll see an output like this:
```

ID : xxx , Name : xxx : Succeeded
```

> Now let's try to remove/delete previously submitted requests. You'll need to update a few things in the request below:

Reviewer: ```suggestion
Now, let's remove a previously submitted request. You'll need to update a few things in the code below:
```

* Replace `<your_key>` with your Speech service subscription key. This information is available in the **Overview** tab for your resource in the [Azure portal](https://aka.ms/azureportal).
* Replace `<region>` with the region where your Speech resource was created (for example: `eastus` or `westus`). This information is available in the **Overview** tab for your resource in the [Azure portal](https://aka.ms/azureportal).
* Replace `<synthesis_id>` with the value returned in the previous request.

> [!NOTE]
> Requests with a status of ‘Running’/'Waiting' cannot be removed or deleted.

Run this command:

```console
python voice_synthesis_client.py – delete -key <your_key> -region <Region> -synthesisId <synthesis_id>
```

> You will get the output like this:

Reviewer: ```suggestion
You'll see an output like this:
```
